viewing_party/party.py

```diff
@@ -1,23 +1,205 @@
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data = {"watchlist": [{"title":MOVIE_TITLE_1,"genre":GENRE_1,"rating":RATING_1}]}
+    return user_data
+
+def watch_movie(janes_data, movie_title1):
+
+    movie_data = {"title": movie_title1 , "genre": GENRE_1, "rating": RATING_1 }
+    for i in janes_data["watchlist"]:
+        if movie_data in janes_data["watchlist"]:
+            janes_data["watchlist"].remove(movie_data)
+            janes_data["watched"].append(movie_data)
+            return janes_data
+    return janes_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(janes_data):
+    total_rating = 0
+    counter = 0
+    if janes_data["watched"] == []:
+        average = 0.0
+        return average
+    else:
+        for val in janes_data.values():
+            for dict in val:
+                for key in dict:
+                    if key == "rating":
+                        total_rating += dict[key]
+                        counter+= 1
+        average = total_rating/counter
+        return average 
+
+def get_most_watched_genre(janes_data):
+    
+    genre_dict ={}
+    
+    if janes_data["watched"] == []:
+        popular_genre = None
+        return popular_genre
+    else:                  
+        for watched_items in janes_data.values():
+            for movie_data in watched_items:
+                key = movie_data["genre"]
+                if key not in genre_dict:
+                    genre_dict[key]= 1 
+                else:
+                    genre_dict[key] +=1
+    popular_genre = max(genre_dict,key=genre_dict.get)
+    return popular_genre
+    
+        
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+def get_unique_watched(amandas_data):
+    amandas_unique_movies =[]
+    list_friends = []
+    list_amandas = []
+
+    for element in amandas_data["friends"]:
+        for movie_data in element["watched"]:
+            list_friends.append(movie_data)
+    
+    
+    for element in amandas_data["watched"]:
+        list_amandas.append(element)
+
+    
+    for element in list_amandas:
+        if element not in list_friends:
+            amandas_unique_movies.append(element)
+
+    return amandas_unique_movies
+
+def get_friends_unique_watched(amandas_data):
+    friends_unique_movies =[]
+    list_friends = []
+    list_amandas = []
+
+    for element in amandas_data["friends"]:
+        for movie_data in element["watched"]:
+                list_friends.append(movie_data)
+    
+    
+    for element in amandas_data["watched"]:
+            list_amandas.append(element)
+  
+
+    for element in list_friends:
+        if element not in list_amandas:
+            if element not in friends_unique_movies:

```

Comment:
Sequential if statements like this can be combined with an 'and'.
```suggestion
        if element not in list_amandas and element not in friends_unique_movies:

```